providers/loading.tsx

```diff
@@ -0,0 +1,19 @@
+import { createContext, Dispatch, SetStateAction, useState } from "react";
+import Loading from "../components/loading";
+
+let setStateAction: Dispatch<SetStateAction<boolean>>;
+
+const LoadingContext = createContext(setStateAction);
+
+function LoadingContextProvider({ children }) {
+  const [loading, setLoading] = useState<boolean>(false);
+
+  return (
+    <LoadingContext.Provider value={setLoading}>
+      <Loading show={loading} />
+      {children}
+    </LoadingContext.Provider>
+  );
+}
+
+export { LoadingContextProvider, LoadingContext };
```

Comment:
You can be more expressive in this return statement, so consumers know whats what:

```ts
export { LoadingContextProvider, isLoading, setLoading: LoadingContext }
```

This would then be consumed (on other components other than app)

```ts
import {isLoading, setLoading} from '../providers/loading-provider';

setLoading(true) // shows
console.asset(isLoading === true, `isLoading value should be true`)
setLoading(false); // hides
console.assert(isLoading === false, `isLoading value should be false`)
```

Reply:
`export { LoadingContextProvider, isLoading, setLoading: LoadingContext }`

have a problem `setLoading: LoadingContext`.

- solution suggestion 

> `const [isloading, setLoading] = useState<boolean>(false);`
> `const LoadingContext = createContext(isloading);`

> `export { LoadingContextProvider, isloading, setLoading: LoadingContext };`


Can be?